services/ingestion/src/pdf_ingestion.py

```diff
@@ -126,18 +125,20 @@ def load_page_data(img_dir: str, current_obj: Mapping[T, T]) -> Mapping[T, T]:
     return page_data
 
 
-def load_pdf_metadata(pdf_path: str, current_obj: Mapping[T, T]) -> Mapping[T, T]:
+def load_pdf_metadata(pdf_path, current_obj):
     """
     Load the pdf metadata
     """
     pdf_name = os.path.basename(pdf_path)
     # Df maps coordinates to unicode, limit is the dimensions of the pdf
     df, limit = parse_pdf(pdf_path)
-    df = df.to_dict()
-    # Hack here: throw this df into json and back
-    df = json.dumps(df)
-    df = json.loads(df)
-    limit = list(limit)
+    if df is not None:
+        df = df.to_dict()
+        # Hack here: throw this df into json and back
+        df = json.dumps(df)
+        df = json.loads(df)
+        limit = list(limit)
+    df = limit = None
```

Comment:
Is this intentional?

Reply:
Good catch, theres supposed to be an else there.

Reply:
Nope, there's a missing else there, good catch.